api/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -166,6 +166,74 @@ type SingleStepInstrumentation struct {
 	// (Requires Agent 7.52.0+ and Cluster Agent 7.52.0+)
 	// +optional
 	LanguageDetection *LanguageDetectionConfig `json:"languageDetection,omitempty"`
+
+	// Targets is a list of targets to apply the auto instrumentation to. The first target that matches the pod will be
+	// used. If no target matches, the auto instrumentation will not be applied. Full config key:
+	// apm_config.instrumentation.targets
```

Comment:
We should add stuff like this so it's clear the agent version this supports:
```suggestion
	// apm_config.instrumentation.targets
	// (Requires Cluster Agent 7.64.0+)
```